pkg/virtctl/vmexport/vmexport_test.go

```diff
@@ -642,13 +650,84 @@ var _ = Describe("vmexport", func() {
 			Expect(err).To(HaveOccurred())
 		})
 	})
-})
 
-func handleVMExportDelete(client *kubevirtfake.Clientset, name string) {
-	client.Fake.PrependReactor("delete", "virtualmachineexports", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
-		delete, ok := action.(testing.DeleteAction)
-		Expect(ok).To(BeTrue())
-		Expect(delete.GetName()).To(Equal(name))
-		return true, nil, nil
+	Context("Port-forward", func() {
+		var (
+			orgHttpFunc virtctlvmexport.HandleHTTPRequestFunc
+		)
+
+		BeforeEach(func() {
+			orgHttpFunc = virtctlvmexport.HandleHTTPRequest
+			testInit(http.StatusOK)
+		})
+
+		AfterEach(func() {
+			virtctlvmexport.HandleHTTPRequest = orgHttpFunc
+			testDone()
+		})
+
+		It("VirtualMachineExport download fails when using port-forward with an invalid port", func() {
+			vme := utils.VMExportSpecPVC(vmexportName, metav1.NamespaceDefault, "test-pvc", secretName)
+			vme.Status = utils.GetVMEStatus([]exportv1.VirtualMachineExportVolume{
+				{
+					Name:    "no-test-volume",
+					Formats: utils.GetExportVolumeFormat(server.URL, exportv1.KubeVirtRaw),
+				},
+			}, secretName)
+			utils.HandleSecretGet(kubeClient, secretName)
+			utils.HandleVMExportGet(vmExportClient, vme, vmexportName)
+			utils.HandleServiceGet(kubeClient, fmt.Sprintf("virt-export-%s", vme.Name), 321)
+			utils.HandlePodList(kubeClient, fmt.Sprintf("virt-export-pod-%s", vme.Name))
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(commandName, virtctlvmexport.DOWNLOAD, vmexportName, setflag(virtctlvmexport.PORT_FORWARD_FLAG, "5432"), setflag(virtctlvmexport.OUTPUT_FLAG, "disk.img"))
+			err := cmd()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).Should(Equal("Service virt-export-test-vme does not have a service port 443"))
+		})
+
+		It("VirtualMachineExport download with port-forward fails when the service doesn't have a valid pod ", func() {
```

Comment:
This test and the one above has 2 differences.
1. It doesn't call HandlePodList (to get the error you want)
2. Checks for a different error.
Maybe it would make sense to try and use a different export volume format in this test just to exercise some of that? My first instinct was can't we make this into a table with some parameters since so much of the setup is the same. Or maybe move this into a context, and have a JustBeforeEach that does the common setup, and then in the It do the slight difference.